opentelemetry-exporter-geneva/geneva-uploader/src/bench.rs

```diff
@@ -43,4 +46,218 @@ mod benchmarks {
+                        ),
+                    ),
+                },
+                _ => AnyValue {
+                    value: Some(
+                        opentelemetry_proto::tonic::common::v1::any_value::Value::BoolValue(
+                            rng.gen_bool(0.5),
+                        ),
+                    ),
+                },
+            };
+
+            log.attributes.push(KeyValue {
+                key,
+                value: Some(value),
+            });
+        }
+
+        log
+    }
+
+    /*
+    - Criterion benchmarks (encode_log_batch, OtlpEncoder):
+        Results: (WSL2 Linux, 62 GiB RAM, 8-core/16-thread Intel CPU, kernel 6.6.36.3)
+        - Attribute count scaling (10 log records each):
+            - 0 attrs:   ~10.3 µs/op
+            - 4 attrs:   ~15.7 µs/op
+            - 8 attrs:   ~24.3 µs/op
+            - 16 attrs:  ~44.1 µs/op
+
+        - Batch size scaling (each log with 4 attributes):
+            - 1 log:     ~1.63 µs/op
+            - 10 logs:   ~16.4 µs/op
+            - 100 logs:  ~161 µs/op
+            - 1000 logs: ~1.59 ms/op
+
+        - Schema cache effectiveness (100 log records, each with 4 attributes, all same schema):
+            - ~165 µs/op
+
+        - Mixed event names (100 logs, 3 different event names, 4 attributes each):
+            - ~168 µs/op
+
+        - To run: $cargo test --release encode_log_batch_benchmark -- --nocapture --ignored
+    */
+    #[test]
+    #[ignore = "benchmark on crate private, ignored by default during normal test runs"]
+    fn encode_log_batch_benchmark() {
+        let mut criterion = Criterion::default();
+        let encoder = OtlpEncoder::new();
+        let metadata = "namespace=benchmark/eventVersion=Ver1v0";
+
+        // Benchmark 1: Different numbers of attributes
+        let mut group = criterion.benchmark_group("encode_log_batch_attributes");
+        for num_attrs in [0, 4, 8, 16].iter() {
+            group.throughput(Throughput::Elements(*num_attrs as u64));
+            group.bench_with_input(
+                BenchmarkId::new("attributes", num_attrs),
+                num_attrs,
+                |b, &num_attrs| {
+                    let mut rng = StdRng::seed_from_u64(42);
+                    let logs: Vec<LogRecord> = (0..10)
+                        .map(|i| {
+                            if num_attrs == 0 {
+                                create_basic_log_record(1_700_000_000_000_000_000 + i)
+                            } else {
+                                create_log_with_attributes(
+                                    1_700_000_000_000_000_000 + i,
+                                    num_attrs,
+                                    &mut rng,
+                                )
+                            }
+                        })
+                        .collect();
+
+                    b.iter(|| {
+                        let res = black_box(encoder.encode_log_batch(logs.iter(), metadata));
```

Comment:
We want to `black_box` the input here, right?

```suggestion
                        let res = encoder.encode_log_batch(black_box(logs.iter()), black_box(metadata));
```